sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -68,7 +68,7 @@ abstract class PlanExpression[T <: QueryPlan[_]] extends Expression {
  */
 abstract class SubqueryExpression(
     plan: LogicalPlan,
-    outerAttrs: Seq[Expression],
+    val outerAttrs: Seq[Expression],
```

Comment:
nit: this needs to touch all the subclasses, which is a bit messy. How about we just add a new method in this class?
```
def isCorrelated: Boolean = outerAttrs.nonEmpty
```